src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -1457,6 +1465,8 @@ - (void)test017_TestMTRDeviceBasics
                 XCTAssertNotNil(eventDict[MTREventTimestampDateKey]);
             }
         }
+    };
+    delegate.onReportEnd = ^() {
```

Comment:
Hmm.  Pre-existing, but I think at the end of this test when we are doing `[self waitForExpectations:@[ resubscriptionExpectation ] timeout:60];` (around line 1601 after this PR) we should in fact be waiting until report end, not just reachability, I think.  Followup to fix that, please?

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/29365